src/java.desktop/share/classes/javax/swing/JInternalFrame.java

```diff
@@ -1305,6 +1305,7 @@ public JDesktopPane getDesktopPane() {
     /**
      * Sets the <code>JDesktopIcon</code> associated with this
      * <code>JInternalFrame</code>.
+     * Setting <code>null</code> icon is unsupported and may throw NPE if used.
```

Comment:
The field itself is protected and can be changed by the user.

Reply:
I guess that is why we are putting up this statement, no so that user thinks twice before making it null? Should we make the field private?

Reply:
We cannot make it private, we can only update the spec there as well.

Reply:
done